sasdata/transforms/NDrebin.py

```diff
@@ -0,0 +1,687 @@
+
+    Note that this does lose some information from the underlying data,
+        as you are essentially averaging multiple measurements into one bin.
+
+    :data: The data at each point
+    :coords: The locations of each data point, same size of data
+        plus one more dimension with the same length as the
+        dimensionality of the space (Ndim)
+    :data_errs: Optional, the same size as data, the uncertainties on data
+    :axes: The axes of the coordinate system we are binning
+        into. Defaults to diagonal (e.g. (1,0,0), (0,1,0), and 
+        (0,0,1) for 3D data). A list of Ndim element vectors
+    :upper: The upper limits along each axis. Defaults to the largest
+        values in the data if no limits are provided.
+        A 1D list of Ndims values.
+    :lower: The lower limits along each axis. Defaults to the smallest
+        values in the data if no limits are provided.
+        A 1D list of Ndims values.
+    :step_size: The size of steps along each axis. Supercedes
+        num_bins. A list of length Ndim.
+    :num_bins: The number of bins along each axis. Superceded by
+        step_size if step_size is provided. At least one of step_size
+        or num_bins must be provided.
+    :fractional: Whether to perform fractional binning or not. Defaults 
+        to false.
+        -If false, measurements are binned into one bin,
+        the one they fall within. Roughly a "nearest neighbor"
+        approach.
+        -If true, fractional binning will be applied, where
+        the value of a measurement is distributed to its 2^Ndim
+        nearest neighbors weighted by proximity. For example, if
+        a point falls exactly between two bins, its value will be
+        given to both bins with 50% weight. This is roughly a
+        "linear interpolation" approach. Tends to do better at 
+        reducing sharp peaks and edges if data is sampled unevenly.
+        However, this is roughly 2^Ndim times slower since you have
+        to address each bin 2^Ndim more times.
+
+
+    Returns: binned_data, bin_centers_list
+    :binned_data: has size num_bins and is NDimensional, contains
+        the binned data
+    :bin_centers_list: is a list of 1D vectors, contains the
+        axes of the binned data. The coordinates of bin [i,j,k]
+        is given by 
+        bin_centers_list[0][i]*axes[i]+bin_centers_list[1][j]*axes[j]+
+        bin_centers_list[0][k]*axes[k]
+    :binned_data_errs: has size num_bins and is NDimensional, contains
+        the propagated errors of the binned_data
+    :bins_list: is a list of 1D vectors, is similar to bin_centers_list,
+        but instead contains the edges of the bins, so it is 1 longer
+        in each dimension
+    :step_size: is a list of Ndims numbers, contains the step size
+        along each dimensino
+    :num_bins: is a list of Ndims numbers, contains the number
+        of bins along each dimension
+
+
+    An example call might be:
+    # test syntax 1
```

Comment:
Adding `.. code-block::` will help with sphinx rendering of the example script.

https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-code-block